src/app/containers/ArticleTimestamp/shouldLastUpdatedTimestampBeShown.js

```diff
@@ -0,0 +1,33 @@
+import { isLastRelative, isSameDay, isToday } from './helpers';
+
+const hasBeenUpdated = (timeDifferenceMinutes, minutesTolerance) =>
+  timeDifferenceMinutes > minutesTolerance;
+
+const publishedAndUpdatedToday = (firstPublished, lastPublished) =>
+  isToday(firstPublished) && isSameDay(firstPublished, lastPublished);
+
+const publishedAndUpdatedOnSameDayAndNotRelative = (
+  firstPublished,
+  lastPublished,
+) => isSameDay(firstPublished, lastPublished) && !isLastRelative(lastPublished);
+// Could this be reconsidered as publishedAndUpdatedOnDifferentDaysAndLastUpdatedNotRelative
+// !isSameDay
+
+const shouldLastUpdatedTimestampBeShown = ({
+  minutesTolerance,
+  firstPublished,
+  lastPublished,
+}) => {
+  const timeDifferenceMinutes = (lastPublished - firstPublished) / 1000 / 60;
+  return (
+    hasBeenUpdated(timeDifferenceMinutes, minutesTolerance) &&
+    (publishedAndUpdatedToday(firstPublished, lastPublished) ||
+      !publishedAndUpdatedOnSameDayAndNotRelative(
+        firstPublished,
+        lastPublished,
+      ))
```

Comment:
Could we maybe move these out of the return statement and into variables?

```
const isUpdated = hasBeenUpdated(timeDifferenceMinutes, minutesTolerance);
const isPublishedAndUpdatedToday = publishedAndUpdatedToday(firstPublished, lastPublished);
const isPublishedAndUpdatedOnSameDayNotRelative = publishedAndUpdatedOnSameDayAndNotRelative(firstPublished, lastPublished);

return (isUpdated && (isPublishedAndUpdatedToday || !isPublishedAndUpdatedOnSameDayNotRelative));

```

Reply:
Nice, have done - as well as some general variable renaming and reconfiguring